quinn-proto/src/connection/streams.rs

```diff
@@ -172,17 +180,17 @@ impl Streams {
         &mut self,
         id: StreamId,
         buf: &mut [u8],
-    ) -> Result<Option<(usize, bool)>, ReadError> {
+    ) -> Result<Option<(usize, bool, WantTransmit)>, ReadError> {
```

Comment:
What I don't like here, is that if `MAX_STREAM_DATA` would change to the same thing, we would return `WantTransmit, WantTransmit`.

And thereby I get to what I don't think is perfect yet: The naming. 
Should it be a different newtype per frame-type? `WantTransmitMaxData` + `WantTransmitMaxStreamData`? At least makes it obvious in tuples what is what.

Another idea is making `WantTransmit` generic, and embedded the frame type, like `WantTransmit<MaxData>`. But that could be overengineered.

Another option is to simply use structs for these return values, instead of tuples.

Reply:
3 items is my threshold for replacing structs with tuples, so I'm on board with that regardless. It might also be nice to disambiguate `WantTransmit`; a phantom type parameter using a frame type would be more pleasant than a bunch of functionally identical handwritten newtypes, for all that it's a bit cheeky. No strong opinion on whether that's really needed though.

Reply:
I now added a dedicated result structs for all those operations